lib/streamlit/elements/space.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2018-2022 Streamlit Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import cast, TYPE_CHECKING, Literal, Optional
+
+from streamlit.errors import StreamlitAPIException
+from streamlit.proto.Space_pb2 import Space as SpaceProto
+
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+
+SIZE_SMALL = "small"
+SIZE_MEDIUM = "medium"
+SIZE_LARGE = "large"
+
+
+class SpaceMixin:
+    def space(
+        self, amount: Optional[Literal["small", "medium", "large"]] = None
+    ) -> "DeltaGenerator":
+        """
+        st.space()
+        """
+        size = None
+
+        if amount is None:
+            size = "small"
+        else:
+            size = amount
```

Comment:
If you cast back to `Literal["small", "medium", "large"]`, you can keep the `size.lower()` call, which I think is useful for people who don't use type checkers. To avoid having to type out this verbose type again, you can use a type alias:

`Size = Literal["small", "medium", "large"]`

This can then be used in both places the type is used. The cast would look like:

`size = cast(Size, size.lower())`